lib/restforce.rb

```diff
@@ -60,9 +60,11 @@ module Tooling
   MatchesMultipleError= Class.new(ResponseError)
   EntityTooLargeError = Class.new(ResponseError)
 
+  require 'restforce/error_codes'
+
   module ErrorCode
     def self.const_missing(constant_name)
-      const_set constant_name, Class.new(ResponseError)
+      raise ArgumentError, "No error code defined for #{constant_name}"
```

Comment:
On second thoughts, I think we should take a slightly more risk-averse approach here and continue to define constants which are not found, but log some kind of warning saying that an unexpected error was encountered and that the user should report it to us at https://github.com/restforce/restforce/issues/new/choose.

That will ensure that we don't accidentally break someone's code. In a future version, once we are more confident that we have all of the errors defined, we could remove this.

Reply:
There are two scenarios where code might break:

1. They reference `Restcode::ErrorCode::Something` in their code that isn't defined.
2. The API returns an error code not mapped to an existing constant.

I think (1) is not a problem - the failure will be obvious as soon as the library is updated.

(2) can be addressed by removing the reliance on `const_get`/`const_missing` in `exception_class_for_error_code` and just defining the constant there.

Reply:
(1) is true assuming that they are not doing some kind of weird const lookup - which is probs it a fair assumption to make. I think we should address (2) by doing a hash lookup (as I describe in my other comment), and if the hash lookup fails:

1. Define the constant
2. Log a warning asking the user to create an issue reporting the missing error